modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,441 @@
+//If you do not agree to this license, do not download, install,
+//copy or use the software.
+//
+//
+//                          License Agreement
+//               For Open Source Computer Vision Library
+//                       (3-clause BSD License)
+//
+//Copyright (C) 2000-2015, Intel Corporation, all rights reserved.
+//Copyright (C) 2009-2011, Willow Garage Inc., all rights reserved.
+//Copyright (C) 2009-2015, NVIDIA Corporation, all rights reserved.
+//Copyright (C) 2010-2013, Advanced Micro Devices, Inc., all rights reserved.
+//Copyright (C) 2015, OpenCV Foundation, all rights reserved.
+//Copyright (C) 2015, Itseez Inc., all rights reserved.
+//Third party copyrights are property of their respective owners.
+//
+//Redistribution and use in source and binary forms, with or without modification,
+//are permitted provided that the following conditions are met:
+//
+//  * Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//
+//  * Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+//  * Neither the names of the copyright holders nor the names of the contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+//This software is provided by the copyright holders and contributors "as is" and
+//any express or implied warranties, including, but not limited to, the implied
+//warranties of merchantability and fitness for a particular purpose are disclaimed.
+//In no event shall copyright holders or contributors be liable for any direct,
+//indirect, incidental, special, exemplary, or consequential damages
+//(including, but not limited to, procurement of substitute goods or services;
+//loss of use, data, or profits; or business interruption) however caused
+//and on any theory of liability, whether in contract, strict liability,
+//or tort (including negligence or otherwise) arising in any way out of
+//the use of this software, even if advised of the possibility of such damage.
+
+/*****************************************************************************************************************\
+*   The interface contains the main descriptors that will be implemented in the descriptor class                  *
+\******************************************************************************************************************/
+
+#include "precomp.hpp"
+#include <stdint.h>
+#ifndef _OPENCV_DESCRIPTOR_HPP_
+#define _OPENCV_DESCRIPTOR_HPP_
+#ifdef __cplusplus
+
+namespace cv
+{
+    namespace stereo
+    {
+        enum { Dense_Census, Sparse_Census, StarCensus};
+        enum {ClassicCenterSymetricCensus, ModifiedCenterSymetricCensus};
+        enum {StandardMct,MeanVariation};
+        enum {SSE, NonSSE};
```

Comment:
Conventionally, OpenCV doesn't allow switching between different optimization backends with run-time parameters. Instead, the optimizations are turned on by default if they are available using `#ifdef HAVE_SSE` machinery. I guess, it's safe to completely remove it for now, since we don't have any SSE in this PR anyway. When you decide to add the optimizations, please, see how other functions do this with "HAVE_SSE" macros.
Conventionally, OpenCV doesn't allow switching between different optimization backends with run-time parameters. Instead, the optimizations are turned on by default if they are available using `#ifdef HAVE_SSE` machinery. I guess, it's safe to completely remove it for now, since we don't have any SSE in this PR anyway. When you decide to add the optimizations, please, see how other functions do this with "HAVE_SSE" and the like macros.